dts: nordic: add auxpll bindings

Hi-Im-David opened the pull request:
Added new auxpll macro definitions for frequency selection. Will be used further when AUXPLL driver is used on more devices.

Requires PR to be merged into NRFX and then hal_nordic

Hi-Im-David committed:
dts: bindings: clock: add nordic,nrf-auxpll

Introduce Nordic NRF AUXPLL binding include.

Signed-off-by: David Jewsbury <david.jewsbury@nordicsemi.no>

Hi-Im-David committed:
dts: vendor: nordic: nrf54h20 update canpll to use auxpll binding

Added new binding for macros of possible auxpll frequency settings.

Will be used in future products also.

Signed-off-by: David Jewsbury <david.jewsbury@nordicsemi.no>

masz-nordic commented on include/zephyr/dt-bindings/clock/nrf-auxpll.h:
```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Apache-2.0

masz-nordic commented on include/zephyr/dt-bindings/clock/nrf-auxpll.h:
```diff
@@ -0,0 +1,16 @@
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_NRF_AUXPLL_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_NRF_AUXPLL_H_
+
+#define NRF_AUXPLL_FREQ_MIN        0
+#define NRF_AUXPLL_FREQ_AUDIO_44K1 15309
+#define NRF_AUXPLL_FREQ_USB24M     32768
+#define NRF_AUXPLL_FREQ_AUDIO_48K  39845
+#define NRF_AUXPLL_FREQ_MAX        65535
```
Please add verification of those values against ones exposed in HAL.

Hi-Im-David committed:
drivers: clock_control_nrf_auxpll: Add checking of frequency in dts and mdk.

Add build time checks for valid frequency setting in devicetree and
check if MDK aligns with known frequencies in dts bindings.

Signed-off-by: David Jewsbury david.jewsbury@nordicsemi.no
